components/match2/wikis/brawlstars/match_group_input_custom.lua

```diff
@@ -326,7 +341,27 @@ function mapFunctions.getTournamentVars(map)
 end
 
 function mapFunctions.getParticipantsData(map)
-	local participants = map.participants or {}
+	local participants = {}
+
+	local maximumPickIndex = 0
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		local pickIndex = 1
+		while String.isNotEmpty(map['t' .. opponentIndex .. 'p' .. pickIndex]) do
+			local brawlerRaw = map['team' .. opponentIndex .. 'pick' .. pickIndex]
+				or map['t' .. opponentIndex .. 'p' .. pickIndex]
+			local brawler = BrawlerNames[string.lower(brawlerRaw)]
+			if not brawler then
+				error('Unsupported brawler input ' .. brawlerRaw)
+			end
```

Comment:
Extract into a separate function, code duplication with 298-301.
Look into extracting the entire loop into a separate function, would be preferable (used by both getParticipantsData and getExtraData)